lib/browser/evaluate.js

```diff
@@ -56,3 +56,33 @@ exports.evaluate = function evaluate() {
       throw new Error(invocation);
   }
 };
+
+exports.evaluateAsync = function evaluateAsync() {
+  const args = _.toArray(arguments);
+  let clientFunction = args.pop();
+
+  const invocation =
+    'evaluateAsync(clientFunction) - requires (Function) clientFunction';
+  assert.truthy(invocation, clientFunction);
+
+  if (typeof clientFunction !== 'function') {
+    throw new Error(invocation);
+  }
+
+  clientFunction = `var args = [].slice.call(arguments), done = args.pop();
+    try {
+      var result = (${clientFunction}).apply(this, args);
+      Promise.resolve(result)
+      .then(result => done({reason: null, value: result}))
+      .catch(err => done({reason: JSON.stringify(err)}));
+    } catch(err) {
+      done({reason: JSON.stringify(err)})
+    }`;
+
+  return this.executeAsync(clientFunction, args).then(result => {
+    if (result.reason) {
+      throw JSON.parse(result.reason);
+    }
```

Comment:
the response will already be JSON-encoded, so assuming you do something as shown above, you could then rehydrate it with something like:
```js
.then(({ reason, value }) => {
  if (reason) throw Object.assign(new Error(reason.message), reason);
  return value;
})